byterun/caml/misc.h

```diff
@@ -297,9 +292,10 @@ void caml_gc_message (int, char *, uintnat);
 extern uintnat caml_runtime_warnings;
 int caml_runtime_warnings_active(void);
 
-/* Memory routines */
-
-char *caml_aligned_malloc (asize_t bsize, int, void **);
+/* Deprecated aliases */
+#define caml_aligned_malloc caml_stat_alloc_aligned_noexc
+#define caml_strdup caml_stat_strdup
```

Comment:
What's the point of the macro definitions here?
You've replaced `caml_strdup` with `caml_stat_strdup` everywhere inside the OCaml code, and for outside code the macros are  not visible, because `CAML_INTERNALS` is not defined 😕

Reply:
This looks like an oversight in the way the headers were updated.

@fdopen - this feature has been added for 4.06 which is still in beta. The reports are of course helpful, but it would be nice if they could be worded in a somewhat more friendly manner, and preferably as a Mantis ticket.

Reply:
[MPR#7652](https://caml.inria.fr/mantis/view.php?id=7652)

Reply:
@cakeplus: would you be able to comment on whether hiding `caml_strdup` when `CAML_INTERNALS` is not set is intentional (and should be documented as a breaking change) or something to revert?

Reply:
Hello! It surely wasn't intentional, as some of the C stubs in OPAM make use of this function. I'll take a closer look at this on the weekends, and will prepare a patch.

Reply:
Subtle ping :-) @cakeplus, note that if you happened to not have time around now, it's just fine, @dra27 or myself can take care of it -- but we need to get it done this week.

Reply:
I've just got my hands on it. One moment... :-)

Reply:
Done: #1442.